### PR TITLE
Refactor status overlay types

### PR DESCRIPTION
**Pull request by warp-core**
**Refactor**

~I like opening PRs that will conflict with PRs I already have open if merged.~

## Summary
Currently, the "type" of a status overlay (flagship, friendly, hostile, neutral, scan) is represented by an "int", without an obvious meaning for each values. To find the meaning, you have to go look at a local array defined somewhere in an Engine method.
This makes it really easy to make mistakes when dealing with this stuff.
This PR introduces an enum class giving each "type" value a name, so it is easier to tell what is going on and easier to identify and correct bugs, or otherwise expand this functionality in the future.
This is a purely "under the hood" change and should not affect how anything looks.

## Testing Done
None.

## Performance Impact
Immeasurable, if any.

### CODE REVIEW

File: source/Engine.cpp [hunk 1]
```diff
@@ -2745,21 +2748,26 @@ void Engine::CreateStatusOverlays()
 		if(it->IsDestroyed())
 			continue;
 
+		static auto FLAGSHIP = Preferences::OverlayType::FLAGSHIP;
+		static auto FRIENDLY = Preferences::OverlayType::ESCORT;
+		static auto HOSTILE = Preferences::OverlayType::ENEMY;
+		static auto NEUTRAL = Preferences::OverlayType::NEUTRAL;
```
Comment (tibetiroka) [inline source/Engine.cpp:2754]: Does something like this work?
```suggestion
		using enum Preferences::OverlayType;
```

Comment (warp-core) [reply source/Engine.cpp:2754]: clang says yes

Comment (TomGoodIdea) [reply source/Engine.cpp]: what compiler is it?

Comment (warp-core) [reply source/Engine.cpp]: GCC 10.3, apparently.
For reference:
CI job | GCC version
-- | --
Ubuntu 22.04 GL | 11.4
Ubuntu 24.04 GL | 13.2
Windows | 12.2

MacOS uses AppleClang 15 and Windows Clang uses Clang 18.1.8, but those aren't directly as comparable.

Comment (warp-core) [reply source/Engine.cpp]: `using enum` is indeed not supported in gcc versions less than 11.
Near the bottom of the C++20 section: https://gcc.gnu.org/projects/cxx-status.html
https://wg21.link/p1099r5

File: source/Engine.h [hunk 2]
```diff
@@ -125,9 +125,20 @@ class Engine {
...
 
 	class Status {
+	public:
+		enum class Type {
+			FLAGSHIP,
+			FRIENDLY,
+			HOSTILE,
+			NEUTRAL,
+			SCAN,
+			SCAN_OUT_OF_RANGE,
```
[the remark this answers is not in the document]
Comment (warp-core) [reply source/Engine.h:135]: Yes
```suggestion
```

**Review by tibetiroka — APPROVED** [review]

**Pull request merged**